### DISCUSSION TITLE
Optimize INNER JOIN query that utilizes same columns for joining and filtering

### DISCUSSION THREAD

**Discussion by bxh296** [ClickHouse/ClickHouse#54159] [Q&A]
ClickHouse v. 23.7.5

I have a query (produced by Tableau) like 
`SELECT .... FROM table_a AS a INNER JOIN table_b AS b ON a.column_a=b.column_a AND a.column_b=b.column_b WHERE a.column_a in ('val1', 'val2',...) AND a.column_b in ('val3', 'val4')`

The `EXPLAIN` on this query shows that filter condition pushes down only to the `table_a`, thus query scans all rows in the `table_b`, it takes relatively long time (`table_b` ~1B rows) . It would be much more efficient to automatically add the same filter as for the `table_a` to the `table_b` for columns that are both in `JOIN` and in `WHERE` unified by `AND` condition (`column_a` and `column_b` in the example), because `JOIN` is of `INNER` type and in any case these rows from the right `table_b` will be filtered out.

Is it possible to add such a feature to the query analyzer?

**Comment by UnamedRus** [comment]
Yes, it's planned.

Will be done after https://github.com/ClickHouse/ClickHouse/issues/42648 will be production ready
